pkg/operation/botanist/controlplane.go

```diff
@@ -400,7 +400,18 @@ func (b *Botanist) DeploySeedMonitoring() error {
 		values["alertmanager"].(map[string]interface{})["emailConfigs"] = emailConfigs
 	}
 
-	return b.ApplyChartSeed(filepath.Join(common.ChartPath, "seed-monitoring"), fmt.Sprintf("%s-monitoring", b.Shoot.SeedNamespace), b.Shoot.SeedNamespace, nil, values)
+	if err := b.ApplyChartSeed(filepath.Join(common.ChartPath, "seed-monitoring"), fmt.Sprintf("%s-monitoring", b.Shoot.SeedNamespace), b.Shoot.SeedNamespace, nil, values); err != nil {
+		return err
+	}
+
+	// If no mail receivers for the Shoot alerts provided than there is no need
+	// to run the Alertmanager and it can be removed from the control plane.
+	if len(emailConfigs) == 0 {
+		if err := common.DeleteAlertmanager(b.K8sSeedClient, b.Shoot.SeedNamespace, common.AlertManagerStatefulSetName); err != nil {
```

Comment:
What's the need for passing the sts name?

Reply:
Importing `github.com/gardener/gardener/pkg/operation/common` in `pkg/operation/common/utils.go` produce an import cycle. And I wanted to reuse the const.

```
import cycle not allowed
package main
	imports github.com/gardener/gardener/cmd/gardener-controller-manager/app
	imports github.com/gardener/gardener/pkg/controllermanager/controller
	imports github.com/gardener/gardener/pkg/controllermanager/controller/backupinfrastructure
	imports github.com/gardener/gardener/pkg/apis/garden/v1beta1/helper
	imports github.com/gardener/gardener/pkg/operation/common
	imports github.com/gardener/gardener/pkg/operation/common
```

Reply:
https://github.com/gardener/gardener/pull/709#pullrequestreview-200569609